### DISCUSSION TITLE
ArpEntry element code example

### DISCUSSION THREAD

**Discussion by Tagman** [wixtoolset/issues#7583] [Questions]
Hi,

I am stuck trying to install a C++ Runtime prerequisite before I run the project msi file.
WiX is not recognizing the already installed VC Runtime and tries to install it. I am entering the arguments for the <ArpEntry> tag wrong. But I have not found any clear example of where to get the needed information.
Currently, I have this:
`<ExePackage
	      SourceFile="vc_redist.x64.exe">
	      <ArpEntry Id="Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.36.32532" Version="14.36.32532.0" Win64="yes"/>
      </ExePackage>`

This information is directly from the Windows "Add/Remove product" page. But the log shows the following: 

> Detected package: vc_redist.x64.exe, state: Absent, cached: No, install registration state: Absent, cache registration state: Absent

My workaround is to use a conditional variable defined through a registry search, as described in the (outdated) tutorial.
https://www.firegiant.com/wix/tutorial/net-and-net/bootstrapping/

**Comment by robmen** [comment]
`"Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.36.32532"` is the display name for the VC Redist, not its ARP Id.

**Reply by Tagman** [reply]
I imagined, that this will not work. Could you provide me with a hint on where to get the correct ARP Id, please? Is it the Upgrade or ProductCode of it?

**Reply by BMurri** [reply]
On my system it's `{8bdfe669-9705-4184-9368-db9ce581e0e7}` but I don't know if this applies to other variations of the runtime or not.
